core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilterWithJwt.java

```diff
@@ -77,4 +96,61 @@ protected void validateAccessToken(String cookieValue)
         assertThat(claims.get("client_id")).isEqualTo(TRINO_CLIENT_ID);
         assertThat(claims.getIssuer()).isEqualTo("https://localhost:4444/");
     }
+
+    @Test
+    public void testBearerAuthenticatorUserMappingApplied()
+            throws Exception
+    {
+        String token = hydraIdP.getToken(TRINO_CLIENT_ID, TRINO_CLIENT_SECRET, ImmutableList.of(TRINO_AUDIENCE));
+        assertUserIdentityResponse("/username", token, TRINO_CLIENT_USERNAME, true, response -> {
+            assertThat(response.code()).isEqualTo(OK.getStatusCode());
+            assertEquals(response.header("user"), TRINO_CLIENT_USERNAME);
+        });
+    }
+
+    @Test
+    public void testBearerAuthenticatorWithImpersonationAllowedNoUserMapping()
+            throws Exception
+    {
+        String token = hydraIdP.getToken(TRUSTED_CLIENT_ID, TRUSTED_CLIENT_SECRET, ImmutableList.of(TRUSTED_CLIENT_ID));
+        assertUserIdentityResponse("/username", token, "test@trino.org", true, response -> {
+            assertThat(response.code()).isEqualTo(OK.getStatusCode());
+            assertEquals(response.header("user"), "test@trino.org");
+        });
+    }
+
+    @Test
+    public void testBearerAuthenticatorWithImpersonationDeniedNoUserMapping()
```

Comment:
I think there is one test case still missing: similar to this one but with `TRINO_CLIENT_ID` & `TRINO_CLIENT_USERNAME`, should return 200 or something like that.

Reply:
Adde 4'd test: testBearerAuthenticatorWithImpersonationDeniedUserMapping
So we have all cases covered.